/sandbox/code/test_visualization_20250610_113834_913860_test_visualization.py:21: UserWarning: Glyph 36724 (\N{CJK UNIFIED IDEOGRAPH-8F74}) missing from font(s) DejaVu Sans.
  plt.savefig(output_path)
/sandbox/code/test_visualization_20250610_113834_913860_test_visualization.py:21: UserWarning: Glyph 38543 (\N{CJK UNIFIED IDEOGRAPH-968F}) missing from font(s) DejaVu Sans.
  plt.savefig(output_path)
/sandbox/code/test_visualization_20250610_113834_913860_test_visualization.py:21: UserWarning: Glyph 26426 (\N{CJK UNIFIED IDEOGRAPH-673A}) missing from font(s) DejaVu Sans.
  plt.savefig(output_path)
/sandbox/code/test_visualization_20250610_113834_913860_test_visualization.py:21: UserWarning: Glyph 25955 (\N{CJK UNIFIED IDEOGRAPH-6563}) missing from font(s) DejaVu Sans.
  plt.savefig(output_path)
/sandbox/code/test_visualization_20250610_113834_913860_test_visualization.py:21: UserWarning: Glyph 28857 (\N{CJK UNIFIED IDEOGRAPH-70B9}) missing from font(s) DejaVu Sans.
  plt.savefig(output_path)
/sandbox/code/test_visualization_20250610_113834_913860_test_visualization.py:21: UserWarning: Glyph 22270 (\N{CJK UNIFIED IDEOGRAPH-56FE}) missing from font(s) DejaVu Sans.
  plt.savefig(output_path)
/sandbox/code/test_visualization_20250610_113834_913860_test_visualization.py:21: UserWarning: Glyph 31034 (\N{CJK UNIFIED IDEOGRAPH-793A}) missing from font(s) DejaVu Sans.
  plt.savefig(output_path)
/sandbox/code/test_visualization_20250610_113834_913860_test_visualization.py:21: UserWarning: Glyph 20363 (\N{CJK UNIFIED IDEOGRAPH-4F8B}) missing from font(s) DejaVu Sans.
  plt.savefig(output_path)
Traceback (most recent call last):
  File "/sandbox/code/test_visualization_20250610_113834_913860_test_visualization.py", line 21, in <module>
    plt.savefig(output_path)
  File "/usr/local/lib/python3.12/dist-packages/matplotlib/pyplot.py", line 1251, in savefig
    res = fig.savefig(*args, **kwargs)  # type: ignore[func-returns-value]
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/matplotlib/figure.py", line 3490, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "/usr/local/lib/python3.12/dist-packages/matplotlib/backend_bases.py", line 2184, in print_figure
    result = print_method(
             ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/matplotlib/backend_bases.py", line 2040, in <lambda>
    print_method = functools.wraps(meth)(lambda *args, **kwargs: meth(
                                                                 ^^^^^
  File "/usr/local/lib/python3.12/dist-packages/matplotlib/backends/backend_agg.py", line 481, in print_png
    self._print_pil(filename_or_obj, "png", pil_kwargs, metadata)
  File "/usr/local/lib/python3.12/dist-packages/matplotlib/backends/backend_agg.py", line 430, in _print_pil
    mpl.image.imsave(
  File "/usr/local/lib/python3.12/dist-packages/matplotlib/image.py", line 1657, in imsave
    image.save(fname, **pil_kwargs)
  File "/usr/local/lib/python3.12/dist-packages/PIL/Image.py", line 2576, in save
    fp = builtins.open(filename, "w+b")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/sandbox/outputs/visualization_test/scatter_plot_example.png'
